### ISSUE TITLE
Crash when copy sga-block in course structure

### ISSUE THREAD

**Issue by eazaika** [mitodl/edx-sga#252]
Hi, 
on hawthorn.2 you can reproduce this bug:
1. In Studio create block with sga (in any course)
2. Then go to the course structure page (in studio)
3. Then try to copy this block by pushing button "Copy" on the block
4. Getting error
```
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 185, in inner
    return func(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 23, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/edx-platform/cms/djangoapps/contentstore/views/course.py", line 270, in course_handler
    return course_index(request, CourseKey.from_string(course_key_string))
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 23, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 149, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/edx-platform/cms/djangoapps/contentstore/views/course.py", line 657, in course_index
    ) if current_action else None,
  File "/usr/lib/python2.7/contextlib.py", line 24, in __exit__
    self.gen.next()
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/mixed.py", line 1016, in bulk_operations
    yield
  File "/usr/lib/python2.7/contextlib.py", line 24, in __exit__
    self.gen.next()
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/__init__.py", line 192, in bulk_operations
    self._end_bulk_operation(course_id, emit_signals, ignore_case)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/__init__.py", line 286, in _end_bulk_operation
    dirty = self._end_outermost_bulk_operation(bulk_ops_record, structure_key)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/split_mongo/split.py", line 254, in _end_outermost_bulk_operation
    self.db_connection.insert_structure(bulk_write_record.structures[_id], bulk_write_record.course_key)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/modulestore/split_mongo/mongo_connection.py", line 436, in insert_structure
    self.structures.insert(structure_to_mongo(structure, course_context))
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mongodb_proxy.py", line 115, in __call__
    return self.proxied_object(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/pymongo/collection.py", line 540, in insert
    gen(), check_keys, self.uuid_subtype, client)
InvalidDocument: Cannot encode object: u'Staff Graded Assignment'
```
PS: there is no ADVANCED_COMPONENT_TYPES settings in cms/envs/common.py

**Comment by fdns** [comment]
As an update, we are using Ironwood.2 and have the same problem copying any xblock that contains an edx_sga component.

We are exploring the problem, but can't find anything right now. Does someone know where we should look to solve the issue?

**Comment by MHaton** [comment]
Confirming that we're seeing this same issue on our Hawthorn site, but it occurs when you duplicate any unit **containing** the XBlock, not the XBlock itself within a unit.
```
InvalidDocument pymongo.collection in insert
Cannot encode object: u'Staff Graded Assignment'
```

**Comment by tasawernawaz** [comment]
@pdpinch what is the release process for this xblock.

**Comment by HlaThit** [comment]
Hello @pdpinch @eazaika @tasawernawaz , I have the same problem in copying units with SGA. How can I solve this problem? Please help me.

**Comment by pdpinch** [comment]
@HlaThit it's not a good idea to comment on closed tickets. 

This issue should have been fixed in the 0.9.0 release. If you are still seeing the bug and you are using a later release that 0.9.0, please open a new issue. 

It might also be helpful to post this question on discuss.openedx.org, if you haven't already.

**Comment by HlaThit** [comment]
@pdpinch Thank you for your reply
